tests/lib/preseed.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# mount ubuntu cloud image through qemu-nbd and mount
+# critical virtual filesystems (such as proc) under
+# the root of mounted image.
+mount_ubuntu_image() {
+    local CLOUD_IMAGE=$1
+    local IMAGE_MOUNTPOINT=$2
+
+    if ! lsmod | grep nbd; then
+        modprobe nbd
+    fi
+
+    qemu-nbd -c /dev/nbd0 "$CLOUD_IMAGE"
```

Comment:
Ah, that's an interesting question, I have rather blindly applied the pattern I got from cloud team & their helper script; I didn't know either of these tools and wasn't aware kpartx alone can mount img file. I need to investigate. Thanks for the question!

Reply:
Ok, so after some digging it turns out it's kpartx not needed here. qemu-nbd is needed because the image is in qcow2 format. The nbd0p1 device can be mounted directly. Thanks for questioning that part!